### PR TITLE
Migrate NumberInput to use Field component

### PR DESCRIPTION
**Pull request by fleonard**
## What does this do?

If you don't mind checking out the branch and test it might be a good thing cause it has **A LOT** of edge cases 😢

### CODE REVIEW

File: src/TextInput/TextInput.tsx [hunk 1]
```diff
@@ -60,7 +61,7 @@ export const TextInput = forwardRef(function TextInput(
       outlined={outlined}
       value={value}
     >
-      <>
+      <Box flex>
```
Comment (fleonard) [inline src/TextInput/TextInput.tsx:64]: this is needed to fix the trailing icon that's now broken because we remove `flex` from the field content

File: src/NumberInput/NumberInput.tsx [hunk 2]
```diff
@@ -220,109 +214,115 @@ export const NumberInput: FC<NumberInputProps> = ({
...
   border-color: ${({ error }) =>
     theme.colors[`${error ? 'error' : 'outline'}`]};
-  display: flex;
-  align-items: center;
-  height: 32px;
+  color: ${({ error }) => theme.colors[`${error ? 'error' : 'secondary'}`]};
+  font-size: 16px;
+  width: 100%;
+  outline: none;
   cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
```
Comment (mickaelzhang) [inline src/NumberInput/NumberInput.tsx:241]: Not part of your PR but `pointer` is a bit weird for an input? Shouldn't we use `text` instead?

**Review by mickaelzhang — APPROVED** [review]
